lib/ts/recipe/oauth2provider/recipeImplementation.ts

```diff
@@ -312,19 +315,20 @@ export default function getRecipeInterface(
             }
 
             const res = await querier.sendPostRequest(
-                new NormalisedURLPath(`/recipe/oauth2/pub/token`),
-                body,
+                new NormalisedURLPath(`/recipe/oauth/token`),
+                { body, iss: await this.getIssuer({ userContext: input.userContext }) },
```

Comment:
I think we should rename the `body` parameter (let's discuss this in the core PR comments)